sympy/ntheory/primetest.py

```diff
@@ -110,6 +110,31 @@ def mr(n, bases):
     return True
 
 
+def isprimeE(n, b):
+    """Test whether n is prime or an Euler pseudoprime to base b."""
+	if (not mr(n, [b])):
+        return False
+    
+    r = int(r)
```

Comment:
indentation problem and r has not yet been defined. See `mr` for a way to check that n and b are integers by using `as_int`.

Reply:
Yes sir,
Thanks or suggestion Its updated now.